components/encryption/src/manager/mod.rs

```diff
@@ -34,12 +36,32 @@ use crate::{
 
 const KEY_DICT_NAME: &str = "key.dict";
 const FILE_DICT_NAME: &str = "file.dict";
+const FILE_DICT_V2_NAME: &str = "file.dict.v2";
 const ROTATE_CHECK_PERIOD: u64 = 600; // 10min
 
 struct Dicts {
-    // Maps data file paths to key id and metadata. This file is stored as plaintext.
+    base: PathBuf,
+    rotation_period: Duration,
+    // Directories directly under these (`whitelist/<name>`) will be represented using Dir ID. This
+    // makes it possible to rename directory without modifying records of every file.
+    //
+    // The reasons for having a whitelist instead of treating all directories this way:
+    // (1) keep v1 data keys file backward compatible.
```

Comment:
We don't need to consider compatibility for now (may not even in the future). Simple is better.

Reply:
It's about v1 data, because v1 and v2 share the implementation of `DataKeyManager`. To implement a completely new `DataKeyManagerV2` is not economical because not all files should be encoded with dir id (for the reason (2)), they should fallback to the old format.

Reply:
> because not all files should be encoded with dir id

Can you give some examples that will not be encoded with dir id?

Reply:
Local backup export file to an arbitrary user-specified folder; nested folder such as titandb.

Reply:
@tabokie DataKeyManager is referenced in many codes , so I agree it's better to keep DataKeyManager as an entry struct for encryption/decryption.  However, I also think we can simply copy all the current implementation for v1, just wrap under DataKeyManager. And the new implementation can only take effect on v2. It may simplify the logic.
Otherwise it may introduce regression. 
Maybe in the future once the new implementation is battle proved, then we unify the two implementations.

Reply:
@tonyxuqqi Missed your comment yesterday. Implementing two managers looks really bad, you have to trait them because there're so many places use them. But I think it's doable to separate the v2 functionality into a new plugin struct, i.e. if the whitelist is empty (for v1), the plugin is None and we don't create the new key file etc.